samples/ResourceOwnerPasswordFlow/Startup.cs

```diff
@@ -0,0 +1,176 @@
+using System.IO;
+using System.Text;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Hosting;
+using Microsoft.Framework.ConfigurationModel;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Runtime;
+
+using ResourceOwnerPasswordFlow.Providers;
+
+
+/// <summary>
+/// Configure the resource owner password credential flow. 
+/// 1. The Relying Party sends a request to the Identity Provider.
+/// 2. The Identity Provider responds with an Access Token. 
+/// 3. The Relying Party saves that Access Token. 
+/// 4. The Relying Party sends a request (with the Access Token) to the Resource Server. 
+/// 5. The Resource Server validates the Access Token.
+/// 6. If it's valid, the Resource Server responds with the requested resources.
+/// </summary>
+
+
+namespace ResourceOwnerPasswordFlow
+{
+    public class Startup
+    {
+        public IConfiguration Config { get; set; }
+
+        public Startup(IHostingEnvironment hostEnv, IApplicationEnvironment appEnv)
+        {
+            // When running in a different environment than the localhost,
+            // be sure to update either config.json or environmental variables
+            // with appropriate OAuth and OpenId configuration values
+            Config = new Configuration(appEnv.ApplicationBasePath)
+                .AddJsonFile("config.json")
+                .AddEnvironmentVariables();
+        }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddAuthentication();
+        }
+
+        public void Configure(IApplicationBuilder app, IRuntimeEnvironment environment)
+        {
+            // HACK: for Azure web apps, 
+            // the try-catch block lets us see errors that occur during configuration
+            try
+            {
+                app.UseErrorPage();
+
+                // Allow serving of .html files in wwwroot
+                app.UseStaticFiles();
+
+                // Add a new middleware validating access tokens issued by the server.
+                // This middleware is associated with the Resource Server
+                app.UseOAuthBearerAuthentication(options =>
+                {
+                    options.AutomaticAuthentication = true;
+                    options.Authority = Config.Get("OAuth:Authority");
+                    options.Audience = Config.Get("OAuth:Audience");
+
+                    // if the audience is null or empty, then don't validate it
+                    options.TokenValidationParameters.ValidateAudience = Config.Get("OAuth:Audience") != null;
+                });
+
+                // Add a new middleware issuing tokens.
+                // This middleware is associated with the Identity Provider            
+                var credentials = CreateSigningCredentials();
+                app.UseOpenIdConnectServer(options =>
+                {
+                    options.Issuer = Config.Get("OpenId:Issuer");
```

Comment:
Hmm. That's cool. So, do you recommend explicitly setting it or just leaving it?